src/app/gm-tools/feature/enter-event-scores-page.component.ts

```diff
@@ -0,0 +1,201 @@
+      }}
+    </p>
+
+    @if (eventTeams(); as teams) {
+      <p-table [value]="teams" [scrollable]="true">
+        <ng-template pTemplate="header">
+          <tr>
+            <th>Team</th>
+            <th>Pos</th>
+            <th>Score</th>
+          </tr>
+        </ng-template>
+        <ng-template
+          pTemplate="body"
+          [joshiesStronglyTypedTableRow]="teams"
+          let-team
+        >
+          <tr>
+            <td>
+              <div class="flex align-items-center gap-2">
+                {{ team.seed }}
+                <p-avatarGroup styleClass="mr-2">
+                  @for (
+                    participant of team.participants;
+                    track participant.participant_id
+                  ) {
+                    <p-avatar
+                      [image]="participant.avatar_url"
+                      size="large"
+                      shape="circle"
+                    />
+                  }
+                </p-avatarGroup>
+              </div>
+            </td>
+            <td>
+              {{ team.position }}
+            </td>
+            <td>
+              <p-inputNumber
+                [(ngModel)]="team.score"
+                [showButtons]="true"
+                buttonLayout="horizontal"
+                [step]="1"
+                [allowEmpty]="false"
+                incrementButtonIcon="pi pi-plus"
+                decrementButtonIcon="pi pi-minus"
+                inputStyleClass="w-full font-semibold text-right"
+                class="w-full"
+                styleClass="w-full"
+                (ngModelChange)="calculatePositions()"
+              />
+            </td>
+          </tr>
+        </ng-template>
+      </p-table>
+
+      <!-- Submit Button -->
+      <p-button
+        label="Submit Bet"
+        styleClass="w-full mt-2"
+        (onClick)="confirmSubmit()"
+        [loading]="submitting()"
+      />
+    } @else {
+      <p-skeleton height="30rem" />
+    }
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export default class EnterEventScoresPageComponent {
+  private readonly eventService = inject(EventService);
+  private readonly confirmationService = inject(ConfirmationService);
+  private readonly messageService = inject(MessageService);
+  private readonly activatedRoute = inject(ActivatedRoute);
+  private readonly router = inject(Router);
+
+  private readonly teams = this.eventService.eventTeamsWithParticipantInfo;
+  readonly eventForThisRound = this.eventService.eventForThisRound;
+
+  readonly submitting = signal<boolean>(false);
+
+  readonly eventTeams = computed(() => {
+    this.eventService.eventForThisRound()?.lower_scores_are_better;
+    return this.teams()
+      ?.filter(
+        (team) =>
+          team.event_id ===
+          (this.eventService.eventForThisRound()?.id ?? false),
+      )
+      .map((team) => {
+        return { ...team, score: 0, position: 1 };
+      });
+  });
+
+  calculatePositions() {
```

Comment:
Mutating the value returned from a Signal (in `[(ngModel)]="team.score"` and in `calculatePositions()`) seems dangerous. Angular can _usually_ handle mutating values like this, but it _could_ cause unexpected behavior.

It might be safer if the calculated positions were in their own variable ([like the `FormGroup` in the End Round page](https://github.com/c-o-l-i-n/joshies/blob/main/src/app/gm-tools/feature/end-round-page.component.ts#L152)), or if they were somehow part of the `eventTeams` computed Signal.